cli/config.py

```diff
@@ -39,19 +45,62 @@ def _load_secrets_config(self, path: str=aztk.utils.constants.DEFAULT_SECRETS_PA
             self._merge_dict(secrets_config)
 
     def _merge_dict(self, secrets_config):
-        # Ensure all necessary fields are provided
+        service_principal = secrets_config.get('servicePrincipal')
+        if service_principal:
+            self.service_principal_tenant_id = service_principal.get('tenantid')
+            self.service_principal_client_id = service_principal.get('clientid')
+            self.service_principal_credential = service_principal.get('credential')
+            self.batch_account_resource_id = service_principal.get('batchaccountresourceid')
+            self.storage_account_resource_id = service_principal.get('storageaccountresourceid')
+
+        shared_key = secrets_config.get('sharedKey')
         batch = secrets_config.get('batch')
+        storage = secrets_config.get('storage')
+
+        if shared_key and (batch or storage):
+            raise aztk.error.AztkError(
+                "Shared keys must be configured either under 'sharedKey:' or under 'batch:' and 'storage:', not both.")
+
+        if shared_key:
+            self.batch_account_name = shared_key.get('batchaccountname')
+            self.batch_account_key = shared_key.get('batchaccountkey')
+            self.batch_service_url = shared_key.get('batchserviceurl')
+            self.storage_account_name = shared_key.get('storageaccountname')
+            self.storage_account_key = shared_key.get('storageaccountkey')
+            self.storage_account_suffix = shared_key.get('storageaccountsuffix')
+
         if batch:
             self.batch_account_name = batch.get('batchaccountname')
             self.batch_account_key = batch.get('batchaccountkey')
             self.batch_service_url = batch.get('batchserviceurl')
 
-        storage = secrets_config.get('storage')
         if storage:
             self.storage_account_name = storage.get('storageaccountname')
             self.storage_account_key = storage.get('storageaccountkey')
             self.storage_account_suffix = storage.get('storageaccountsuffix')
 
+        if self.batch_account_resource_id or self.storage_account_resource_id:
+            if not self.service_principal_tenant_id:
+                raise aztk.error.AztkError("Service principal auth missing tenant_id")
+            if not self.service_principal_client_id:
+                raise aztk.error.AztkError("Service principal auth missing client_id")
+            if not self.service_principal_credential:
+                raise aztk.error.AztkError("Service principal auth missing credential")
+        else:
+            if self.service_principal_tenant_id or self.service_principal_client_id or \
+               self.service_principal_credential:
+                raise aztk.error.AztkError(
+                    "Service principal auth incomplete: must have tenantid, clientid, credential "
+                    "and at least one of batchaccountresourceid and storageaccountresourceid.")
```

Comment:
here shouldn't the message be updated? batch_account resource id is required right?